app/src/main/java/app/revanced/integrations/returnyoutubedislike/ReturnYouTubeDislike.java

```diff
@@ -222,48 +190,82 @@ public static void sendVote(Vote vote) {
-
-            userId = ReturnYouTubeDislikeApi.registerAsNewUser(); // blocks until network call is completed
-            if (userId != null) {
-                SettingsEnum.RYD_USER_ID.saveValue(userId);
-            }
+        String userId = SettingsEnum.RYD_USER_ID.getString();
+        if (userId != null) {
             return userId;
         }
+
+        userId = ReturnYouTubeDislikeApi.registerAsNewUser(); // blocks until network call is completed
+        if (userId != null) {
+            SettingsEnum.RYD_USER_ID.saveValue(userId);
+        }
+        return userId;
     }
 
-    private static void updateDislike(AtomicReference<Object> textRef, Integer dislikeCount) {
-        SpannableString oldSpannableString = (SpannableString) textRef.get();
+    /**
+     * @return if the video likes are visible, and the dislike count was added to the UI
+     */
+    private static boolean updateDislike(AtomicReference<Object> textRef, boolean isSegmentedButton, int dislikeCount) {
+        var dislikeString = formatDislikeCount(dislikeCount);
 
-        // Parse the buttons string.
-        // If the button is segmented, only get the like count as a string
-        var oldButtonString = oldSpannableString.toString();
-        if (segmentedButton) oldButtonString = oldButtonString.split(" \\| ")[0];
+        SpannableString oldSpannableString = (SpannableString) textRef.get();
 
-        var dislikeString = formatDislikes(dislikeCount);
-        SpannableString newString = new SpannableString(
-                segmentedButton ? (oldButtonString + " | " + dislikeString) : dislikeString
-        );
+        SpannableString newDislikeString;
+        if (isSegmentedButton) {
+            // If the button is segmented, then parse out the like count as a string
+            var oldButtonString = oldSpannableString.toString();
+            oldButtonString = oldButtonString.split(" \\| ")[0];
+
+            // YouTube creators can hide the like count on a video,
+            // and the like count appears as a device language specific string that says 'Like'
+            // check if the first character is not a number
+            if (!Character.isDigit(oldButtonString.charAt(0))) {
+                // likes are hidden.
+                // RYD does not provide usable data for these types of videos,
+                // and the API returns bogus data (zero likes and zero dislikes)
+                //
+                // example video: https://www.youtube.com/watch?v=UnrU5vxCHxw
+                // RYD data: https://returnyoutubedislikeapi.com/votes?videoId=UnrU5vxCHxw
+                //
+                // discussion about this: https://github.com/Anarios/return-youtube-dislike/discussions/530
+
+                //
+                // RYD browser plugin shows the dislikes as a localized translated string of "disabled by owner"
+                // ReVanced could do the same, but for now just leave as-is
+                //
+                return false;
```

Comment:
If returned here, the user would not get any feedback on wether the dislikes are disabled, the server is down, or wether the patch has an issue. Some kind of feedback should be given such as the string in the comment above, or something shorter such as "OFF"

Reply:
I changed the "Likes" string to say:
   "Likes Hidden by Creator"

It makes the button somewhat large, but it more accurately describes the situation.
These type of videos are not common, so I don't think the extra UI space is not a big concern.

Reply:
Shouldn't it say "dislikes hidden by creator"? (I do not remember, but if the creator hides the dislike count, is that already written in the likes string?)

Reply:
The only indication that likes are hidden by the creator, is no number of likes is shown.
If the 'Likes' label says "Dislikes Hidden by Creator", but then users might question "But where is the likes count?"

If the [cosmetic layout bug](https://github.com/revanced/revanced-integrations/issues/118) is resolved, then I think the ideal is to show "Dislikes Hidden by Creator" in the 'Dislike' label.  But right now that's not possible since both likes and dislikes are jammed into the single 'Likes' label.

Maybe show "Likes and Dislikes Hidden by Creator"?
It's very wordy, but it would completely convey the situation.

Reply:
"Likes and Dislikes Hidden by Creator" is too wordy, and the dislike button is almost off the screen.

Reply:
[Likes Hidden by Creator](https://freeimage.host/i/HfAHiVR)

[Likes and Dislikes Hidden by Creator](https://freeimage.host/i/HfAHsPp)

Reply:
Can the creator even "only" hide the likes?

Reply:
The creator can only hide the likes.  

Dislikes are always hidden by YouTube.

Reply:
In that case, just saying "Hidden" would suffice while also be short enough to not be obnoxiously wide.